lib/signalfx/tracing/sfx_logger.rb

```diff
@@ -0,0 +1,68 @@
+require 'logger'
+require 'fileutils'
+
+module SignalFx
+  module Tracing
+    module Logging
+
+      LOG_LEVELS = {
+          'unknown' => Logger::UNKNOWN,
+          'fatal' => Logger::FATAL,
+          'error' => Logger::ERROR,
+          'warn' => Logger::WARN,
+          'info' => Logger::INFO,
+          'debug' => Logger::DEBUG
+      }.freeze
+
+      DEFAULT_LOG_PATH = '/var/log/signalfx/signalfx-ruby-tracing.log'
+      DEFAULT_SHIFT_AGE = 5
+      DEFAULT_SHIFT_SIZE = 1048576
+
+      def self.create (log_path = ENV['SIGNALFX_LOG_PATH'] || DEFAULT_LOG_PATH,
+                       sfx_shift_age = ENV['SIGNALFX_LOG_SHIFT_AGE'] || DEFAULT_SHIFT_AGE,
+                       sfx_shift_size = ENV['SIGNALFX_LOG_SHIFT_SIZE'] || DEFAULT_SHIFT_SIZE)
+
+        log_level = ENV['SIGNALFX_LOG_LEVEL'].downcase if ENV['SIGNALFX_LOG_LEVEL']
+        if log_path == 'STDOUT'
+          @logger = Logger.new(STDOUT)
+        elsif log_path == 'STDERR'
+          self.create_stderr_logger(file_exists=false)
+        else
+          file_exists = File.exist?(log_path)
+          if file_exists
```

Comment:
If the process has permission to write to the parent dir, it will create a log file so this check is incorrect.  Resulting errors should dictate whether to use stderr instead.

Reply:
I thought we agreed that the process should not be the one creating the log file/directory?

Reply:
No, we shouldn't be making the directory since that's not the behavior of the std logger.  The file will be created by it if necessary (we don't need to worry about that).